packages/react-components/src/components/xy-output-component.tsx

```diff
@@ -576,23 +654,37 @@ export class XYOutputComponent extends AbstractTreeOutputComponent<AbstractOutpu
     }
 
     private getTimeForX(x: number): bigint {
-        if (!this.chartRef.current?.chartInstance) {
-            return BigInt(0);
+        if (this.isBarPlot) {
+            if (!this.chartRef.current) {
+                return BigInt(0);
+            }
+        } else {
+            if (!this.chartRef.current?.chartInstance) {
+                return BigInt(0);
+            }
         }
         const offset = this.props.viewRange.getOffset() ?? BigInt(0);
         const duration = this.props.viewRange.getDuration();
+        const chartWidth = this.getChartWidth();
         const time = this.props.viewRange.getStart() - offset +
-            BIMath.round(x / this.chartRef.current.chartInstance.width * Number(duration));
+            BIMath.round(x / chartWidth * Number(duration));
         return time;
     }
 
     protected getXForTime(time: bigint): number {
-        if (!this.chartRef.current?.chartInstance) {
-            return 0;
+        if (this.isBarPlot) {
+            if (!this.chartRef.current) {
+                return 0;
+            }
+        } else {
+            if (!this.chartRef.current?.chartInstance) {
+                return 0;
+            }
```

Comment:
Since we're not using the chart anymore, maybe we can skip these checks altogether?

Reply:
Done.